### PR TITLE
Alerting: Disable unified alerting by default in Enterprise Grafana

### PR DESCRIPTION
**Pull request by yuri-tceretian**
**What this PR does / why we need it**:
In #42200 we enabled the Unified alerting by default for new installations as well as installations that do not use legacy alerting. 
This PR amends the changes by alerting the behavior if Grafana is in the Enterprise mode. According to the product requirements, in that mode, Grafana 8 alerting is supposed to be disabled by default. 
|  `GF_ALERTING_ENABLED` |  `GF_UNIFIED_ALERTING_ENABLED`  | Result |
| ---------------------- | ------------------------------- | ------ |
| `true` | `true` | Error |
| `true` | `false` | Legacy |
| `true` | not set | Legacy |
| not set | `true` | UA |
| not set | `false` | Legacy |
| not set | not set | Legacy |
| `false` | `true` | UA |
| `false` | `false` | Alerting disabled |
| `false` | not set | Alerting disabled |

### CODE REVIEW

File: pkg/setting/setting_test.go [hunk 1]
```diff
@@ -433,12 +434,14 @@ func TestAlertingEnabled(t *testing.T) {
...
 		legacyAlertingEnabled  string
 		featureToggleSet       bool
+		isEnterprise           bool
 		verifyCfg              func(*testing.T, Cfg, *ini.File)
 	}{
 		{
 			desc:                   "when legacy alerting is enabled and unified is disabled",
 			legacyAlertingEnabled:  "true",
 			unifiedAlertingEnabled: "false",
+			isEnterprise:           rand.Int63()%2 == 0,
```
Comment (JohnnyQQQQ) [inline pkg/setting/setting_test.go:444]: This might be a stupid question, but why do we use `rand.Int63` here?

Comment (yuri-tceretian) [reply pkg/setting/setting_test.go:444]: to indicate that the value does not affect anything. Alternative would be a test case that covers the combination, which does not make any sense imo.

Comment (JohnnyQQQQ) [reply pkg/setting/setting_test.go:444]: Can we replace it by a const named `doesntMatterForTest` or something like this and document it there?

**Review by pkolyvas — APPROVED** [review]
Thanks Yuri.

File: pkg/setting/setting_test.go (hunk 1 above)
Comment (yuri-tceretian) [reply pkg/setting/setting_test.go]: Good idea. Done!

**Review by JohnnyQQQQ — APPROVED** [review]
LGTM

**Pull request merged**